src/middlewared/middlewared/plugins/pool.py

```diff
@@ -1730,12 +1731,72 @@ def configure_resilver_priority(self):
 
         if data['devices']:
             job.check_pipe("input")
-            args = [job.pipes.input.r, data['passphrase'], data['devices']]
+            key = job.pipes.input.r
         else:
-            args = []
+            key = None
+
+        passfile = None
+        if key and data.get('passphrase'):
+            encrypt = 2
+            passfile = tempfile.mktemp(dir='/tmp/')
+            with open(passfile, 'w') as f:
+                os.chmod(passfile, 600)
+                f.write(data['passphrase'])
+        elif key:
+            encrypt = 1
+        else:
+            encrypt = 0
+
+        pool_name = data.get('name') or pool['name']
+        scrub_id = pool_id = None
+        try:
+            pool_id = await self.middleware.call('datastore.insert', 'storage.volume', {
+                'vol_name': pool_name,
+                'vol_encrypt': encrypt,
+                'vol_guid': data['guid'],
+            })
+            pool = await self.middleware.call('pool.query', [('id', '=', pool_id)], {'get': True})
+            if encrypt > 0:
+                if not os.path.exists(GELI_KEYPATH):
+                    os.mkdir(GELI_KEYPATH)
+                try:
+                    key.seek(0)
```

Comment:
Why? And why we just `pass` if input stream is not seekable?

Reply:
It was an artifact for the legacy UI. Fixed, thanks!